### PR TITLE
Silver Weapon Refactor

### PR DESCRIPTION
**Pull request by Kyogon**
## About The Pull Request

Refactors the silver weapon code, assigning all silver weapons a var that calls on a proc to be used. The code is no longer assigned specifically to the silver dagger.

Or in Layman's terms, it makes it so we can now assign things as silver weapons, and they will work against werewolves and vampires. 
This means that the Elvish Saber, being made of silver, will now work against those two.

However, I also nerfed silver, removing the stun, knockdown, and fire effect on hit. It now purely locks regeneration of the werewolf/vampire it hits, since we all know that is all silver needs to do, and with silver soon becoming more available, this change is needed.

This code refactor also opens the path to creating new silver weapons in the future. If you are a spriter and have something in mind, now is the time. After all, a true silver sword for the holy would be a sight to behold..

-------------------------------------------------------------------------------------------------------------------------------------

**UPDATE:** After feedback and suggestions and a bit of brainstorming with Sigma (who reminded me that the silver curse debuff didn't do what I thought) I made it so silver against werewolves and vampires applies a debuff that gives -2 to all stats and blocks healing and regeneration from their abilities for 45 seconds

Also I fixed werewolves being able to infect people while not transformed. You're welcome!

## Why It's Good For The Game

Cleaner and more modular code that allows for bigger expansion on the silver weapon arsenal.

Nerfing the very overpowered effects of silver on those antags also opens the way to allowing more silver to flow in the town, without making life a living hell for those antags.

### CODE REVIEW

**Comment by MaSvedish** [comment]
I really think the gutting of most of the silver is a bad idea especially with how easy werewolves can bite at people [and floor]. I think just either removing the stun, or the knockdown, and keeping one of those and the fire effect is a good idea.

**Comment by Kyogon** [comment]
> I really think the gutting of most of the silver is a bad idea especially with how easy werewolves can bite at people [and floor]. I think just either removing the stun, or the knockdown, and keeping one of those and the fire effect is a good idea.

I think instead of avoiding nerfing the silver we should just make it so *only transformed bites infect*

**Comment by Kyogon** [comment]
> One small thing against this, although less of a deal breaker and more to keep in mind, is that Silver in I believe all cases (except daggers at the moment but that'll be rectified) is the Best Version of a weapon, because elves. This will make already powerful weapons even more powerful and pro-town.
> 
> Not necessarily a bad thing, but something to keep in mind.

The elvish dagger already counted as a silver weapon, this refactor just mainly brings the sabre into the fold because it was supposed to be beforehand. The nerf to silver is to compensate for possibly introducing better weapons, and right now I am just waiting for silver weapon sprites to introduce more, in which case I will probably have to touch the elvish items to only be found and not made, or to bring them in line with current damage values.

**Comment by Kyogon** [comment]
(Also to note, I got the greenlight from Pantsu to direct some work to the Mountains, which will have Elvish ruins as well)

**Comment by Lutowski** [comment]
love the refactor, opens up more options
not sure on the nerfs themselves (because I don't do combat all that often), I'd recommend meeting a halfway point from where it's currently at so it's not that cataclysmic of a change and see whether that improves anything on live
